src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt

```diff
@@ -0,0 +1,328 @@
+      eventType isEqualTo "CREATE"
+      prisonCode isEqualTo MOORLAND
+      courtDescription isEqualTo "Derby Justice Centre"
+      courtCode isEqualTo DERBY_JUSTICE_CENTRE
+      courtBooking isBool true
+      createdByPrison isBool false
+      preLocationKey isEqualTo moorlandLocation.key
+      preStartTime isEqualTo LocalTime.of(9, 0)
+      preEndTime isEqualTo LocalTime.of(10, 0)
+      mainLocationKey isEqualTo moorlandLocation.key
+      mainDate isEqualTo yesterday()
+      mainStartTime isEqualTo LocalTime.of(10, 0)
+      mainEndTime isEqualTo LocalTime.of(11, 0)
+      preLocationKey isEqualTo moorlandLocation.key
+      postStartTime isEqualTo LocalTime.of(11, 0)
+      postEndTime isEqualTo LocalTime.of(12, 0)
+    }
+  }
+
+  @Test
+  @Transactional
+  fun `should migrate probation booking`() {
+    whereaboutsApi().stubGetVideoBookingToMigrate(
+      2,
+      VideoBookingMigrateResponse(
+        videoBookingId = 2,
+        offenderBookingId = 2,
+        prisonCode = WERRINGTON,
+        courtCode = BLACKPOOL_MC_PPOC,
+        probation = true,
+        createdBy = "MIGRATION PROBATION USER",
+        madeByTheCourt = true,
+        comments = "Migrated probation comments",
+        main = AppointmentLocationTimeSlot(2, 2.daysAgo(), LocalTime.of(10, 0), LocalTime.of(11, 0)),
+        cancelled = false,
+        events = listOf(
+          VideoBookingEvent(
+            eventId = 1,
+            eventTime = 3.daysAgo().atStartOfDay(),
+            eventType = VideoLinkBookingEventType.CREATE,
+            comment = "Probation booking create event comments",
+            courtCode = BLACKPOOL_MC_PPOC,
+            preStartTime = null,
+            preEndTime = null,
+            mainStartTime = 2.daysAgo().atTime(10, 0),
+            mainEndTime = 2.daysAgo().atTime(11, 0),
+            postStartTime = null,
+            postEndTime = null,
+            prisonCode = WERRINGTON,
+            createdByUsername = "MIGRATION PROBATION CREATE USER",
+            courtName = null,
+            madeByTheCourt = true,
+            preLocationId = null,
+            mainLocationId = 2,
+            postLocationId = null,
+          ),
+        ),
+      ),
+    )
+
+    prisonSearchApi().stubPostGetPrisonerByBookingId(2, "DEF123", WERRINGTON)
+
+    UUID.randomUUID().also { dpsLocationId ->
+      nomisMappingApi().stubGetNomisToDpsLocationMapping(2, dpsLocationId.toString())
+      locationsInsidePrisonApi().stubGetLocationById(dpsLocationId, moorlandLocation2)
+    }
+
+    videoBookingRepository.findAll().filter(VideoBooking::isProbationBooking) hasSize 0
+
+    inboundEventsService.process(MigrateVideoBookingEvent(2))
+
+    val migratedProbationBooking = videoBookingRepository.findAll().single(VideoBooking::isProbationBooking)
+
+    with(migratedProbationBooking) {
+      bookingType isEqualTo "PROBATION"
+      statusCode isEqualTo StatusCode.ACTIVE
+      probationMeetingType isEqualTo "UNKNOWN"
```

Comment:
Still need to sort out the data to back up 'UNKNOWN' if that is what we are going with.